src/luxonis_ml/utils/filesystem.py

```diff
@@ -72,22 +78,26 @@ def full_path(self) -> str:
         return f"{self.protocol}://{self.path}"
 
     def init_fsspec_filesystem(self) -> Any:
-        """Returns fsspec filesystem based on protocol"""
+        """Returns fsspec filesystem based on protocol."""
         if self.protocol == "s3":
             # NOTE: In theory boto3 should look in environment variables automatically but it doesn't seem to work
-            return fsspec.filesystem(
+            fs = fsspec.filesystem(
                 self.protocol,
                 key=environ.AWS_ACCESS_KEY_ID,
                 secret=environ.AWS_SECRET_ACCESS_KEY,
                 endpoint_url=environ.AWS_S3_ENDPOINT_URL,
             )
         elif self.protocol == "gcs":
             # NOTE: This should automatically read from GOOGLE_APPLICATION_CREDENTIALS
-            return fsspec.filesystem(self.protocol)
+            fs = fsspec.filesystem(self.protocol)
         elif self.protocol == "file":
-            return fsspec.filesystem(self.protocol)
+            fs = fsspec.filesystem(self.protocol)
```

Comment:
Would there be some weird behavior if we use `file` protocol and also `cache_options` is not `None`?

Reply:
I briefly tested it and it should probably work fine, but I'm now ignoring the cache settings when protocol is `"file"` to be safe.